Resources/public/js/views/services/ez-trashviewservice.js

```diff
@@ -223,6 +223,66 @@ YUI.add('ez-trashviewservice', function (Y) {
         },
 
         /**
+         * Restores trash items
+         *
+         * @method _restoreItems
+         * @protected
+         * @param {Object} e restoreItems event facade
+         * @param {Array} e.trashItems  List of trashItems to be restored
+         */
+        _restoreItems: function (e) {
+            var loadOptions = {api: this.get('capi')},
+                tasks = new Y.Parallel(),
+                service = this,
+                notificationIdentifier = "restoreTrashItems",
+                restoreCount = 0;
+
+            this._notify(
+                'Restoring trash item(s)',
+                notificationIdentifier,
+                'started',
+                0
+            );
+
+            Y.each(e.trashItems, function (trashItem) {
+                trashItem.restore(loadOptions, tasks.add(function (error) {
+                    var trashItemId = trashItem.get('id');
+
+                    if (error) {
+                        service._notify(
+                            "An error occurred while restoring trash item with Id: " + trashItemId
+                            + ". For the moment item's whose original location has been deleted can not be restored",
+                            notificationIdentifier + "-error",
```

Comment:
you should also add the trash item id in the identifier otherwise there will be only one notification even if several items were not restored.